dask/utils.py

```diff
@@ -625,3 +630,83 @@ def insert(tup, loc, val):
     L = list(tup)
     L[loc] = val
     return tuple(L)
+
+
+def infer_storage_options(urlpath, storage_options=None):
+    """ Infer storage options from URL path and merge it with existing storage
+    options.
+
+    Parameters
+    ----------
+    path: str or unicode
+        Either local absolute file path or URL (hdfs://namenode:8020/file.csv)
+    storage_options: dict (optional)
+        Its contents will get merged with the infered information from the
+        given path
+
+    Returns
+    -------
+    Storage options dict.
+
+    >>> so = infer_storage_options('/mnt/datasets/test.csv')
+    >>> so.pop('protocol')
+    'file'
+    >>> so.pop('path')
+    '/mnt/datasets/test.csv'
+    >>> so
+    {}
```

Comment:
These docstrings might be easier to follow if you display the dict directly

```python
>>> infer_storage_options('/mnt/datasets/test.csv')  # doctest: +SKIP
{'protocol': 'file', 'path': '/mnt/datasets/test.csv'}
```

I suspect that you're not doing this so that you can use this as a test.  I think that doctests are documentation first and tests second.  We can always have actual unittests to verify correctness.